File: LeetCode/problems/67-add-binary/67-add-binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string ans = "";

        int X = a.length()-1;
        int Y = b.length()-1;
        int C = 0;

        while( X >= 0 || Y >= 0 || C){
            if(X >= 0){
                C += a[X--]-'0';
            }
            if(Y >= 0){
                C += b[Y--]-'0';
            }
            ans = to_string(C%2) + ans;
            C = C>>1;
        }

        return ans;
    }
};

File: LeetCode/problems/67-add-binary/67-add-binary.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function(a, b) {
    let ans = "";

    let X = a.length-1;
    let Y = b.length-1;
    let C = 0;

    while(X >= 0 || Y >= 0 || C == 1){
        C = ( Number( a[X--]??0 ) + Number( b[Y--]??0 ) + C);
        ans = (C%2) + ans;
        C = C>>1;
    }

    return ans;
};
